src/v/compression/stream_zstd.cc

```diff
@@ -33,6 +35,25 @@ static void throw_if_error(size_t rc) {
     }
 }
 
+// workspace and decompression buffer
+static thread_local size_t dctx_workspace_size = 0;
+static thread_local std::unique_ptr<char[], ss::free_deleter> dctx_workspace;
+static thread_local ss::temporary_buffer<char> d_buffer;
+
+void stream_zstd::init_workspace(size_t size) {
+    if (!dctx_workspace) {
+        dctx_workspace_size = ZSTD_estimateDStreamSize(size);
+        dctx_workspace = ss::allocate_aligned_buffer<char>(
+          dctx_workspace_size, 8); // zstd requires alignment
+        if (!dctx_workspace) {
```

Comment:
what do you think of asserting `vassert()` here? i'm still undecided

Reply:
either should work and terminate application.cc initialization. ill change to vassert.